src/config/app-config-endpoints.html

```diff
@@ -1,11 +1,51 @@
 <script>
 
   AppEndpoints = {
-    protocol: 'https://',
-    subdomain: 'subdomain',
-    domain: '.domain.com'
+    apiBase: {
+      protocol: 'https://',
+      subdomain: 'subdomain',
+      apiSuffix: 'api',
+      domain: 'domain.com'
+    }
   };
 
+  function resolveApiBaseUrl() {
+    const config = AppEndpoints.apiBase;
+    let { protocol, port } = window.location;
+    let [ subdomain, ...domain] = window.location.hostname.split('.');
+
+    // Handle vanity domains (no subdomain)
+    if (!domain.includes('.')) {
```

Comment:
This will never be true, domain is an array of items that were split on '.' so it can't contain a '.' any longer...